### PR TITLE
Homeassistant Rockon based on linuxserver.io docker image

### LINKED ISSUE #285: Update Home Assistant rock-on to latest recommended settings
As reported by our forum user **MainrtNr5**, our Home Assistant rock-on needs some updating due to some limitation when using a share not-owned by `root`:
https://forum.rockstor.com/t/beta-cant-access-certain-rock-on-ui-opensuse-leap-linux-5-3-18-lp152-66-default-4-0-6-0/7706/5

In addition to fixing this (by removing the `"uid"` instruction in our json definition, we might also stick to the project's defaults as defined in their installation instructions: https://www.home-assistant.io/installation/linux#platform-installation
```shell script
docker run --init -d \
  --name homeassistant \
  --restart=unless-stopped \
  -v /etc/localtime:/etc/localtime:ro \
  -v /PATH_TO_YOUR_CONFIG:/config \
  --network=host \
  homeassistant/home-assistant:stable
```

When comparing this to our current settings, two points stand out:
1. We should add the `--init` flag
2. We might also prefer to use the `:stable` tag rather than the currently used `:latest`.

### PR DESCRIPTION
**Pull request by Hooverdan96**
Fixes #285  .

### General information on project
This pull request proposes to add a new rock-on for the following project:
- name: Home Assistant
- website: 
- description: updating home assistant rockon switching to custom docker image to allow for non-root usage

### Information on docker image
- docker image: https://hub.docker.com/r/linuxserver/homeassistant
- is an official docker image available for this project?: yes. Based on discussion in Issue #285, switching to linuxserver.io based image (at least until more flexibility in official image is available)

### Checklist
- [X] Passes [JSONlint](https://jsonlint.com) validation
- [X] Entry added to `root.json` in _alphabetical_ order (for new rock-on only)
- [X] `"description"` object lists and links to the docker image used
- [X] `"description"` object provides information on the image's particularities (advantage over another existing rock-on for the same project, for instance)
- [X] `"website"` object links to project's main website

### CODE REVIEW

File: home-assistant-linuxserver.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
+{
+	"Home Assistant Linuxserver.io": {
+		"description": "Open source home automation that puts local control and privacy first. <p>Based on the custom docker image (so non-root user can be used): <a href='https://hub.docker.com/r/homeassistant/home-assistant' target='_blank'>https://hub.docker.com/r/linuxserver/homeassistant</a>. Available for amd64, armv7 & arm64.</p><p>Note that this container uses the net=host option to allow for multi-cast and uPnP functionality that Home Assistant provides.</> ",
```
Comment (FroggyFlox) [inline home-assistant-linuxserver.json:3]:
```suggestion
		"description": "Open source home automation that puts local control and privacy first. <p>Based on the Linuxserver.io docker image (so non-root user can be used): <a href='https://hub.docker.com/r/linuxserver/homeassistant' target='_blank'>https://hub.docker.com/r/linuxserver/homeassistant</a>. Available for amd64, armv7 & arm64.</p><p>Note that this container uses host networking to allow for multi-cast and uPnP functionality that Home Assistant provides.</> ",
```

File: home-assistant-linuxserver.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+				"launch_order": 1,
+				"opts": [
+					[
+						"--net",
+						"host"
+					]
+				],
+				"ports": {
+					"8123": {
+						"description": "Home Assistant UI port. You may need to open the same port on your firewall (protocol: tcp).",
```
Comment (FroggyFlox) [inline home-assistant-linuxserver.json:22]: Maybe we should add a note that it MUST be 8123 (as it's using host networking).

**Review by FroggyFlox — CHANGES_REQUESTED** [review]
Hi @Hooverdan96,

I haven't had time to test it yet, but I noticed a few minor edits to keep in line with our recent harmonization efforts as well as ensuring the correct port is chosen by the user. 

I'll try to get on with testing it as soon as possible.

File: home-assistant-linuxserver.json (hunk 2 above)
Comment (Hooverdan96) [reply home-assistant-linuxserver.json:22]: Good point, since we are enforcing host mode on this, that makes sense.

**Comment by Hooverdan96** [comment]
as discussed in #285 deprecating older Rockon for Home Assistant

**Comment by phillxnet** [comment]
@Hooverdan96 This looks like another candidate to re-base on master, along with hopefully a minor conflict resolution.

**Review by phillxnet — APPROVED** [review]
@Hooverdan96 My apologies for neglecting this pr. And in the process causing you more work by merging your more recent and major pr first. I'll try to be better behaved in the future ;).

Following on from the earlier review of @FroggyFlox where all suggestions look to have been addressed I've given this a quick functional re-test. Thanks for updating according to the new arch tagging in text.

Installed and tested on amd64 and arm64 and Web-UI appeared and functioned as expected. Picked up a LAN attached smart device and the Rock-on could be turned off and on (pre and post reboot) as expected and survived a reboot in the on state.

HomeAssistant-linuxserver

Nice.

**Comment by phillxnet** [comment]
@FroggyFlox & @Hooverdan96 
PR product in production.

**Comment by phillxnet** [comment]
@Hooverdan96 I forgot to mention a harmless side effect of this pull request. We now have an orphaned home-assistant.json.

@FroggyFlox & @Hooverdan96 do we see any reason for me to not just remove this file?

**Comment by Hooverdan96** [comment]
No reason from my side.

**Pull request merged**